python/ctsm/site_and_regional/single_point_case.py

```diff
@@ -0,0 +1,433 @@
+        logger.info("fluse_out: %s", os.path.join(self.output_dir, fluse_out))
+
+        # create 1d coordinate variables to enable sel() method
+        f_in = self.create_1d_coord(
+            fluse_in, "LONGXY", "LATIXY", "lsmlon", "lsmlat"
+        )
+
+        # extract gridcell closest to plon/plat
+        f_out = f_in.sel(lsmlon=self.plon, lsmlat=self.plat, method="nearest")
+
+        # expand dimensions
+        f_out = f_out.expand_dims(["lsmlat", "lsmlon"])
+
+        # specify dimension order
+        f_out = f_out.transpose(u"time", u"cft", u"natpft", u"lsmlat", u"lsmlon")
+
+        # revert expand dimensions of YEAR
+        year = np.squeeze(np.asarray(f_out["YEAR"]))
+        temp_xr = xr.DataArray(year, coords={"time": f_out["time"]}, dims="time", name="YEAR")
+        temp_xr.attrs["units"] = "unitless"
+        temp_xr.attrs["long_name"] = "Year of PFT data"
+        f_out["YEAR"] = temp_xr
+
+        # update attributes
+        self.update_metadata(f_out)
+        f_out.attrs["Created_from"] = fluse_in
+
+        wfile = os.path.join(self.output_dir, fluse_out)
+        # mode 'w' overwrites file
+        f_out.to_netcdf(path=wfile, mode="w", format="NETCDF3_64BIT")
+        logger.info("Successfully created file (fluse_out), %s", wfile)
+        f_in.close()
+        f_out.close()
+
+        # write to user_nl_clm data if specified
+        if self.create_user_mods:
+            with open(os.path.join(user_mods_dir, "user_nl_clm"), "a") as nl_clm:
+                line = "landuse = '${}'".format(os.path.join(USRDAT_DIR, fluse_out))
+                self.write_to_file(line, nl_clm)
+
+    def create_surfdata_at_point(self, indir, file, user_mods_dir):
+        """
+        Create surface data file at a single point.
+        """
+        logger.info("----------------------------------------------------------------------")
+        logger.info(
+            "Creating surface dataset file at %s, %s", self.plon.__str__(), self.plat.__str__())
+
+        # specify file
+        fsurf_in = os.path.join(indir, file)
+        fsurf_out = add_tag_to_filename(fsurf_in, self.tag)
+        logger.info("fsurf_in:  %s", fsurf_in)
+        logger.info("fsurf_out: %s", os.path.join(self.output_dir, fsurf_out))
+
+        # create 1d coordinate variables to enable sel() method
+        f_in = self.create_1d_coord(fsurf_in, "LONGXY", "LATIXY", "lsmlon", "lsmlat")
+
+        # extract gridcell closest to plon/plat
+        f_out = f_in.sel(lsmlon=self.plon, lsmlat=self.plat, method="nearest")
+
+        # expand dimensions
+        f_out = f_out.expand_dims(["lsmlat", "lsmlon"]).copy(deep=True)
+
+        # update the plon and plat to match the surface data
+        # we do this so that if we create user_mods the PTS_LON and PTS_LAT in CIME match
+        # the surface data coordinates - which is required
+        self.plat = f_out.coords["lsmlat"].values[0]
+        self.plon = f_out.coords["lsmlon"].values[0]
```

Comment:
Thanks for catching this!

I agree that this is problematic - I think that the surface data and the landuse data are generally (have to be?) the same resolution so I don't think that would be an issue - though I agree it is still not best practice to modify the self.plon/plat. And for the DATM data the resolution could change. 

I think we should have new member variables (if necessary) as you suggest. Right now the updated plon/plat are used in a separate function that writes the xml commands that you need. So we could either move part of it here and keep those updated lat/lons internal, or make new class variables.

Reply:
I also meant to do this in `create_landuse_at_point` as well but I see that it is not there... Either way I agree it's best to not do it the way I originally did.

Reply:
Notes from discussion:
- [ ] Get rid of this updating of `self.plat` and `self.plon`; instead rely on the updating of the lat & lon points on the surface dataset (and test to confirm that that works now).

Reply:
I explained in our meeting with @billsacks  and @adrifoster that I previously added these lines to update `lsmlat` and `lsmlon` on the extracted surface data to match the specific point instead of the nearest point:

https://github.com/ESCOMP/CTSM/blob/4e6c6a23aa7f07ea18e7356a1202795d17ea6803/python/ctsm/site_and_regional/single_point_case.py#L269-L273

So there is no need for these lines and @adrifoster confirmed it too. 

@adrifoster and I tested the code removing the lines you are pointing out  and it seems like it was working. So these lines are removed now.

Reply:
This is addressed but I kept the comment that @adrifoster added for the clarity.